File: solutions/go/pangram/1/pangram.go
package pangram

func IsPangram(input string) (b bool) {
	m := make(map[rune]bool)
    for _, v := range input {
        if 'A' <= v && v <= 'Z' || 'a' <= v && v <= 'z' {
            m[v] = true
        }
    }
    for offset := rune(0); offset < 26; offset++ {
        if !m['A' + offset] && !m['a' + offset] {
            return
        }
    }
    b = true
    return
}

File: solutions/javascript/pangram/1/pangram.js
const PANGRAM = 0b11_1111_1111_1111_1111_1111_1111
const charIsLower = ch => ch >= 'a' && ch <= 'z'
const charIdx = ch => ch.charCodeAt(0) - 'a'.charCodeAt(0)
export const isPangram = input => PANGRAM === Array.from(input.toLowerCase())
  .reduce((bits, ch) => charIsLower(ch) ? (bits | 1 << charIdx(ch)) : bits, 0)